### PR TITLE
[MTM-54517]allow registration of custom filters

### PR DESCRIPTION
**Pull request by sguhc8y**
This is a related change to `DAST scan` changes to capture request and transform them postman collection. More information can be found here:

[Pynt doc](https://sagportal.sharepoint.com/:w:/r/sites/IoTAnalyticsRnDOps/_layouts/15/Doc.aspx?sourcedoc=%7BD5F02F23-69B0-4345-9F6E-7F6E3A4F79D9%7D&file=%5BStudy%5D%20PYNT%20Dynamic%20API%20Security%20Testing.docx&action=default&mobileredirect=true)

Quality PR: https://github.softwareag.com/IOTA/cumulocity-quality/pull/2488

### CODE REVIEW

**Comment by kalavathig1** [comment]
can you find me ?

File: java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java [hunk 1]
```diff
@@ -455,6 +456,16 @@ public static Client createClient(PlatformParameters platformParameters) {
         return client;
     }
 
+    private static void registerExternalFilters(PlatformParameters platformParameters,ClientConfig config) {
+         if (platformParameters.customFilterSet != null) {
+            synchronized (platformParameters.customFilterSet) {
```
Comment (kaczynskid-sag) [inline java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java]: Why `synchronized`?

File: java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java [hunk 2]
```diff
@@ -436,6 +436,7 @@ public static Client createClient(PlatformParameters platformParameters) {
         config.property(ClientProperties.FOLLOW_REDIRECTS, true);
 
         registerClasses(config);
+        registerExternalFilters(platformParameters,config);
```
Comment (kaczynskid-sag) [inline java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java]:
```suggestion
        registerExternalFilters(platformParameters, config);
```

File: java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java [hunk 3]
```diff
@@ -455,6 +456,16 @@ public static Client createClient(PlatformParameters platformParameters) {
         return client;
     }
 
+    private static void registerExternalFilters(PlatformParameters platformParameters,ClientConfig config) {
```
Comment (kaczynskid-sag) [inline java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java]:
```suggestion
    private static void registerExternalFilters(PlatformParameters platformParameters, ClientConfig config) {
```

File: java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java [hunk 4]
```diff
@@ -455,6 +456,16 @@ public static Client createClient(PlatformParameters platformParameters) {
...
+         if (platformParameters.customFilterSet != null) {
```
Comment (kaczynskid-sag) [inline java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java]: How can it ever be `null`?

File: java-client/src/main/java/com/cumulocity/sdk/client/PlatformParameters.java [hunk 5]
```diff
@@ -77,6 +78,8 @@ public class PlatformParameters implements AutoCloseable {
 
     Set<HttpClientInterceptor> interceptorSet = Collections.newSetFromMap(new ConcurrentHashMap<>());
 
+    Set<ClientRequestFilter> customFilterSet = Collections.newSetFromMap(new ConcurrentHashMap<>());;
```
Comment (kaczynskid-sag) [inline java-client/src/main/java/com/cumulocity/sdk/client/PlatformParameters.java]:
```suggestion
    Set<ClientRequestFilter> customFilterSet = Collections.newSetFromMap(new ConcurrentHashMap<>());
```

File: java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java (hunk 2 above)
Comment (sayanc8y) [reply java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java]: Corrected

File: java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java (hunk 3 above)
Comment (sayanc8y) [reply java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java]: Corrected

File: java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java (hunk 4 above)
Comment (sayanc8y) [reply java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java]: Agree. Remove this check now. Please have a look.

File: java-client/src/main/java/com/cumulocity/sdk/client/PlatformParameters.java (hunk 5 above)
Comment (sayanc8y) [reply java-client/src/main/java/com/cumulocity/sdk/client/PlatformParameters.java]: Corrected

File: java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java (hunk 1 above)
Comment (sayanc8y) [reply java-client/src/main/java/com/cumulocity/sdk/client/RestConnector.java]: Agree, its not required removed now.

**Review by jkot-c8y — COMMENTED** [review]
current version of PR looks OK for me

**Review by karol-c8y — APPROVED** [review]

**Pull request closed without merging**